### ISSUE TITLE
On translating thunderlinks to cbthunderlinks

### ISSUE THREAD

**Issue by ericdanan** [CamielBouchier/cb_thunderlink#6]
This is just an idea, I don't have any issue now. You mentioned earlier you plan to provide a command line tool that would output a file containing $messageid$ / $cblink$ pairs for all messages, that could then be used to convert old thunderlinks to cbthunderlinks should the former become unusable in Thunderbird, but that this approach would not work for converting links in non-text documents such as word or excel files.

Couldn't a solution be to make cb_thunderlink make the translation on the fly when a thunderlink is clicked? That is, suppose thunderlinks stop working at some point. Then at this moment I use your tool to generate that file (with a hard-coded file name in the same folder as cb_thunderlink.exe. Then when cb_thunderlink.exe receives a thunderlink, if that file is there it uses it to pass the corresponding cbthunderlink to Thunderbird instead. And I only need to generate that file once if I switch to creating cbthunderlinks only from then onwards.

Opening thunderlinks from the extension button might be more tricky, but perhaps possible if the extension can somehow send the thunderlink to cb_thunderlink.exe that would then send back the corresponding cbthunderlink (I suppose there should then be a checkbox in the options page to toggle this behavior).

As I said, I just wanted to run the idea by you in case you find it worthwhile. Feel free to close the issue any time.

**Comment by CamielBouchier** [comment]
@ericdanan : this is *exactly* the idea I got as well. For your reference, I dropped the tool to generate the correspondence file [here.](https://github.com/CamielBouchier/cb_thunderdump)

When running that idea through my head, the most difficult part up to now seems that the accompanying python script needs to ' listen'  to the cb_background.js on stdin ánd to a socket (as it does now) for receiving input from the  command line instance. This proves to be less trivial than it sounds in Windows. But when times comes, I will for sure find a solution. 

For now I am happy the way it works for now and with the warm feeling an alternative is available when things would evolve the wrong direction (that is if thunderbird API would make it impossible to search on msg-id).

Camiel B.
